Adjustments for RustyHermit

stlankes opened the pull request:
The interface between `libstd` and the OS changed and some changes are not correctly merged for RustHermit. For instance, the crate `hermit_abi` isn't defined as public, although it provided the socket interface for the application.

In addition, the support of thread::available_parallelism is realized. It returns the number of available processors.

bors commented:
:umbrella: The latest upstream changes (presumably #104455) made this pull request unmergeable. Please [resolve the merge conflicts](https://rustc-dev-guide.rust-lang.org/git.html#rebasing-and-conflicts).

stlankes commented:
I resolved the merge conflict.

bors commented:
:umbrella: The latest upstream changes (presumably #113046) made this pull request unmergeable. Please [resolve the merge conflicts](https://rustc-dev-guide.rust-lang.org/git.html#rebasing-and-conflicts).

stlankes committed:
add support of available_parallelism for target hermit

On RustyHermit, the function `get_processor_count` returns the
number of activated processors.

stlankes committed:
define hermit_abi as public depedenceny

It's exported publicly, so it should not be linted.

stlankes committed:
fix usage of Timespec om the target hermit

stlankes commented:
I resolved the merge conflict again.

stlankes committed:
use latest version of hermit-abi

0.3.0 and 0.3.1 have an issue and will be yanked. Consequently, std
should switch to 0.3.2.

bors commented:
:umbrella: The latest upstream changes (presumably #113276) made this pull request unmergeable. Please [resolve the merge conflicts](https://rustc-dev-guide.rust-lang.org/git.html#rebasing-and-conflicts).

stlankes commented:
I resolved the conflict.

m-ou-se commented:
@bors r+
